File: frontend/src/types/session.ts
export interface Session {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface SessionDetail {
  session: Session;
  messages: ChatMessageItem[];
}

export interface ChatMessageItem {
  id: string;
  role: "user" | "assistant";
  content: string;
  created_at: string;
}

File: backend/app/api/schemas/session.py
from pydantic import BaseModel


class SessionResponse(BaseModel):
    id: str
    title: str
    created_at: str
    updated_at: str


class ChatMessageSchema(BaseModel):
    id: str
    role: str
    content: str
    created_at: str


class SessionDetailResponse(BaseModel):
    session: SessionResponse
    messages: list[ChatMessageSchema]


class UpdateTitleRequest(BaseModel):
    title: str
